src/components/carousel.tsx

```diff
@@ -0,0 +1,63 @@
+"use client";
+
+import { Swiper, SwiperSlide } from "swiper/react";
+import "swiper/css";
+import "swiper/css/pagination";
+import { Autoplay, Pagination } from "swiper/modules";
+import { ReactNode } from "react";
+
+interface CarouselItem {
+  image?: string;
+  text?: ReactNode;
+}
+
+interface CarouselProps {
+  items: CarouselItem[];
```

Comment:
We want this component to be able to transform anything into a carousel, no just images, so `items` will actually be an array of children and then we will pass whatever we want to this parameter.

```suggestion
  items: React.ReactNode[];
```